Dockerfile

```diff
@@ -1,40 +1,114 @@
-FROM python:3.12-slim-bookworm AS requirement-builder
+# ╔═════════════════════════════════════════════════════╗
+# ║                       SETUP                         ║
+# ╚═════════════════════════════════════════════════════╝
+  # globals
+  ARG APP_UID=1000 \
+      APP_GID=1000 \
+      APP_ROOT=/app
 
-WORKDIR /app
 
-RUN pip install --no-cache-dir poetry poetry-plugin-export
+# ╔═════════════════════════════════════════════════════╗
+# ║                       BUILD                         ║
+# ╚═════════════════════════════════════════════════════╝
+  FROM python:3.12-alpine AS requirement-builder
+  ARG APP_ROOT \
+      PIP_ROOT_USER_ACTION=ignore \
+      PIP_BREAK_SYSTEM_PACKAGES=1 \
+      PIP_DISABLE_PIP_VERSION_CHECK=1 \
+      PIP_NO_CACHE_DIR=1
 
-COPY ./pyproject.toml /app
-COPY ./poetry.lock /app
+  RUN set -ex; \
+    pip install poetry poetry-plugin-export
 
-RUN poetry export --without-hashes -f requirements.txt --output requirements.txt
+  COPY ./pyproject.toml / \
+       ./poetry.lock /
 
-FROM python:3.12-slim-bookworm
+  RUN set -ex; \
+    poetry export --without-hashes -f requirements.txt --output requirements.txt;
 
-ENV PYTHONUNBUFFERED=1 \
-    ORG_ID= \
-    UVICORN_HOST=0.0.0.0 \
-    UVICORN_PORT=5000
+  FROM python:3.12-alpine AS bitwarden-sdk
+  # source https://pypi.org/project/bitwarden-sdk/
+  # sadly we need to build the bitwarden-sdk from scratch and even add the missing license file which is still not fixed!
+  ARG APP_ROOT \
+      PIP_ROOT_USER_ACTION=ignore \
+      PIP_BREAK_SYSTEM_PACKAGES=1 \
+      PIP_DISABLE_PIP_VERSION_CHECK=1 \
+      PIP_NO_CACHE_DIR=1
 
-WORKDIR /app
+  RUN set -ex; \
+    apk --no-cache --update add \
+      git \
+      curl \
+      cargo \
+      rust \
+      py3-setuptools \
+      py3-maturin \
+      py3-gpep517 \
+      py3-wheel; \
+    curl -SL https://files.pythonhosted.org/packages/dd/03/11934ae9d668283895286872a7af3de25d324ec9ac86da5a56ac9dc48544/bitwarden_sdk-1.0.0.tar.gz | tar -zxC /; \
+    cd bitwarden_sdk-1.0.0; \
+    curl -SL https://raw.githubusercontent.com/bitwarden/sdk/refs/heads/main/LICENSE --output LICENSE; \
+    gpep517 build-wheel \
+      --wheel-dir .dist \
+      --output-fd 3 3>&1 >&2;
 
-COPY --from=requirement-builder /app/requirements.txt /app
 
-RUN pip install --no-cache-dir -r requirements.txt && \
-    apt update && \
-    apt install -y curl && \
-    apt-get clean autoclean && \
-    apt-get autoremove --yes && \
-    rm -rf /var/lib/{apt,dpkg,cache,log}
+# ╔═════════════════════════════════════════════════════╗
+# ║                       IMAGE                         ║
+# ╚═════════════════════════════════════════════════════╝
+# :: HEADER
+  FROM 11notes/alpine:stable
```

Comment:
What's wrong with alpine?

Reply:
Using a random 3rd party image opens up a can of supply-chain worms. do not recommend.